luigi/scheduler.py

```diff
@@ -96,12 +95,15 @@ def rpc_method(**request_args):
     def _rpc_method(fn):
         # If request args are passed, return this function again for use as
         # the decorator function with the request args attached.
-        fn_args = six.getargspec(fn)
-
-        assert not fn_args.varargs
-        assert fn_args.args[0] == 'self'
-        all_args = fn_args.args[1:]
-        defaults = dict(zip(reversed(all_args), reversed(fn_args.defaults or ())))
+        args, varargs, varkw, defaults, kwonlyargs, kwonlydefaults, ann = inspect.getfullargspec(fn)
+        if kwonlyargs or ann:
+            raise ValueError("Function has keyword-only parameters or annotations"
+                             ", use getfullargspec() API which can support them")
+        (first_arg, *all_args) = args
+
+        assert not varargs
```

Comment:
Should this be checked before unpacking `args`?

Reply:
yes, makes sense
thanks